onnxruntime/core/framework/utils.cc

```diff
@@ -140,10 +139,16 @@ const std::string& GetNodeInputProviderType(const SessionState::NodeInfo& info)
   return required_provider_type;
 }
 
-static Status CopyMLValue(const DataTransferManager& data_transfer_mgr,
-                          const MLValueCopyInfo& copy_info,
-                          const OrtValue& source_mlvalue,
-                          OrtValue& target_mlvalue) {
+// Copy MLValue. Uses DataTransferManager for device copy if necessary. If copy_pairs is provided,
+// src/dst pairs that need a device copy are added to copy_pairs so copying can be batches by the DataTransferManager
+// implementation for performance reasons.
+static Status CopyMLValue(
```

Comment:
When copy_pairs is non-null, this doesn't do any copy. In this case the only value this method provides is the extraction of the tensor from MLValue and the population of the output vector. It might be cleaner to accumulate what you want to copy outside this method and then call the batch CopyTensors? This method should preserve the semantics of copying one MLValue.

Reply:
So if we accumulate outside we need to maintain two separate vectors - one that needs device copy and one that doesn't. That seems a little messy vs. the single check on src vs dst device here with direct copy when possible. Would renaming this method be better instead? It's local to this file so the name can be anything.

Reply:
Renaming it would be fine.